### DISCUSSION TITLE
https://www.spsanderson.com/steveondata/posts/2024-08-05/

### DISCUSSION THREAD

**Discussion by giscus** [spsanderson/steveondata#5] [Comments]
# https://www.spsanderson.com/steveondata/posts/2024-08-05/

https://www.spsanderson.com/steveondata/posts/2024-08-05/

**Comment by skolenik** [comment]
What happens when `k` in Step 3 is not an integer?

A lot of the times systematic samples are used in practice, it would be in the context of unequal probability sampling. The packaged implementation is sampling::UPsystematic().

**Reply by spsanderson** [reply]
For the implementation in the blog it would fail if `k` was not an integer. The reason is that in this instance `k` is every `k_th` element.
